drivers/crypto/crypto_mtls_shim.c

```diff
@@ -189,17 +591,22 @@ static int mtls_query_caps(struct device *dev)
 
 static int mtls_shim_init(struct device *dev)
 {
+	hwrng = device_get_binding(CONFIG_ENTROPY_NAME);
+	if (!hwrng) {
+		return -ENODEV;
+	}
+
 	return 0;
 }
 
 static struct crypto_driver_api mtls_crypto_funcs = {
 	.begin_session = mtls_session_setup,
-	.free_session = mtls_session_free,
+	.free_session = mtls_session_dev_free,
 	.crypto_async_callback_set = NULL,
 	.query_hw_caps = mtls_query_caps,
 };
 
-DEVICE_AND_API_INIT(crypto_mtls, CONFIG_CRYPTO_MBEDTLS_SHIM_DRV_NAME,
+DEVICE_AND_API_INIT(crypto_mtls, CONFIG_CRYPTO_NAME,
```

Comment:
Shouldn't this be changed in https://github.com/zephyrproject-rtos/zephyr/pull/8351/commits/13c522400e9ec80282d3a99bf27fab2821c41e99 ?